docs/modules/ROOT/pages/how-tos/upgrade-3.x-to-4.x.adoc

```diff
@@ -0,0 +1,90 @@
+= Upgrade from 3.x to 4.x
+
+This guide describes the steps to perform an upgrade of the component from version 3.x to 4.x.
+
+The major changes affect the ingress feature and configuration.
+This component now features re-encrypted ingress by default resp. encrypts cluster-internal traffic.
+
+== Step-by-step guide
+
+. Decide where you want to terminate user traffic.
+  You can either terminate the TLS session at the ingress controller and re-encrypt traffic to Keycloak.
+  Or you can pass-through the TLS traffic directly to Keycloak, provided the ingress controller supports TLS pass-through.
+. Decide how you aim to provision and rotate TLS certificates.
+  The cert-manager operator provides easy Let's Encrypt certificate provisioning by default.
+  You can also provision your own CA issued certificate.
+
+The sections below explain more.
+
+=== Changed parameters
+
+* `ingress.secretName` has moved to `ingress.tls.secretName`.
+* `ingress.enabled` changed to `true` by default.
+* `ingress.annotations` received several annotations depending on `tls.provider` and `tls.termination`.
+* `helm_values.extraVolumes` and `helm_values.extraVolumeMounts` got an additional volume/mount from the `tls.secretName` secret.
+* `helm_values.ingress.servicePort` changed to `https` by default.
```

Comment:
Might be worth mentioning more explicitly that users must update their configuration to match these new parameters if they've already configured some of them.